spec/controllers/api/teams_controller_spec.rb

```diff
@@ -7,80 +7,185 @@
 
   let(:bob) { users(:bob) }
   let(:alice) { users(:alice) }
-  let(:nested_models) { %w[folder account] }
+  let(:team1) { teams(:team1) }
+  let(:team2) { teams(:team2) }
+  let!(:team3) { Fabricate(:non_private_team) }
+  let!(:team4) { Fabricate(:non_private_team) }
+  let(:bobs_private_key) { bob.decrypt_private_key('password') }
+  let!(:team3_user) { team3.teammembers.first.user }
 
   context 'GET index' do
     it 'should get team for search term' do
-      login_as(:alice)
-      get :index, params: { 'q': 'team' }, xhr: true
-
-      result_json = data.first
+      login_as(:bob)
 
-      team = teams(:team1)
+      get :index, params: { 'q': 'mail' }, xhr: true
 
-      expect(result_json['attributes']['name']).to eq team.name
-      expect(result_json['id']).to eq team.id.to_s
+      expect(data.size).to be(0)
+      expect(included).to be(nil)
     end
 
     it 'should get all teams for no query' do
-      login_as(:alice)
+      login_as(:bob)
+
       get :index, params: { 'q': '' }, xhr: true
 
+      expect(data.size).to eq(2)
+
       result_json = data.first
 
-      team = teams(:team1)
+      expect(result_json['attributes']['name']).to eq team1.name
+      expect(result_json['id']).to eq team1.id.to_s
+
+      result_json = data.second
 
-      expect(result_json['attributes']['name']).to eq team.name
-      expect(result_json['id']).to eq team.id.to_s
+      expect(result_json['attributes']['name']).to eq team2.name
+      expect(result_json['id']).to eq team2.id.to_s
+
+      folder_relationships_length = data.first['relationships']['folders']['data'].size
+
+      expect(included.size).to be(6)
+      expect(folder_relationships_length).to be(3)
     end
 
     it 'should get a single team if one team_id is given' do
```

Comment:
'returns single team if team_id given' ... make it always as short as possible :)